java/com/engflow/notificationqueue/Client.java

```diff
@@ -0,0 +1,207 @@
+import com.engflow.eventstore.v1.GetInvocationRequest;
+import com.engflow.eventstore.v1.StreamedBuildEvent;
+import com.engflow.notification.v1.*;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+import io.grpc.ManagedChannel;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import io.netty.handler.ssl.SslContextBuilder;
+
+import javax.annotation.Nullable;
+import javax.net.ssl.SSLException;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+class Client {
+
+  public static void main(String[] args) throws Exception {
+    /**
+     * Check for the command line arguments in specific order.
+     * In this case, only the presence of the flags are checked.
+     * Create appropriated validations when needed.
+     */
+    if (!areAllInputsProvided(args)){
+      System.err.println("Please add the following flags: ");
+      System.err.println(" --notification_queue_endpoint");
+      System.err.println(" --queue_name");
+      System.err.println(" --tls_certificate");
+      System.err.println(" --tls_key");
+      return;
+    }
+
+    /**
+     * Channel used by the NotificationQueueGrpc and the EventStoreGrpc stub. It is also possible to
+     * use two separated channels.
+     */
+    ManagedChannel channel = null;
+    try {
+      channel =
+          createChannel(
+              args[0].split("=")[1],
+              "",
+              args[2].split("=")[1],
+              args[3].split("=")[1]);
+
+      pull(channel, args[1]);
+    } finally {
+      if (channel != null) {
+        channel.shutdownNow();
+
+        try {
+          channel.awaitTermination(10, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+          System.out.println("Could not shut down channel within timeout");
+        }
+      }
+    }
+  }
+
+  private static boolean areAllInputsProvided(String[] args){
+    final String[] global_options = {"--notification_queue_endpoint", "--queue_name", "--tls_certificate", "--tls_key"};
+    int checks = 0;
+    for (String predefinedOption : global_options) {
+      for (String givenOption : args) {
+        if (predefinedOption.equals(givenOption.substring(0, predefinedOption.length()))){
+          checks++;
+          break;
+        }
+      }
+    }
+    return checks == 4 ? true : false;
+  }
+  private static void pull(ManagedChannel channel, String queueName) throws InterruptedException {
+    NotificationQueueGrpc.NotificationQueueStub asyncStub = NotificationQueueGrpc.newStub(channel);
+    final CountDownLatch finishLatch = new CountDownLatch(1);
+    StreamObserver<PullNotificationRequest> requestObserver =
```

Comment:
I don't know this API well enough to understand what's happening here. That will likely be the case for other readers too, though. Please add some comments?